.github/workflows/preview_deploy.yml

```diff
@@ -0,0 +1,71 @@
+name: Deploy Preview
+on: 
+  workflow_run:
+    workflows: ["Generate Preview"]
+    types:
+      - completed
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    if: >
+      ${{ github.event.workflow_run.event == 'pull_request' &&
+      github.event.workflow_run.conclusion == 'success' }}
+    steps:
+      - name: 'Download artifact'
+        uses: actions/github-script@v3.1.0
+        with:
+          script: |
+            var artifacts = await github.actions.listWorkflowRunArtifacts({
+               owner: context.repo.owner,
+               repo: context.repo.repo,
+               run_id: ${{ github.event.workflow_run.id }},
+            });
+            var matchPrArtifact = artifacts.data.artifacts.filter((artifact) => {
+              return artifact.name == "pr"
+            })[0];
+            var matchPreviewArtifact = artifacts.data.artifacts.filter((artifact) => {
+              return artifact.name == "preview"
+            })[0];
+            var downloadPr = await github.actions.downloadArtifact({
+               owner: context.repo.owner,
+               repo: context.repo.repo,
+               artifact_id: matchPrArtifact.id,
+               archive_format: 'zip',
+            });
+            var downloadPreview = await github.actions.downloadArtifact({
+               owner: context.repo.owner,
+               repo: context.repo.repo,
+               artifact_id: matchPreviewArtifact.id,
+               archive_format: 'zip',
+            });
+            var fs = require('fs');
+            fs.writeFileSync('${{github.workspace}}/pr.zip', Buffer.from(downloadPr.data));
+            fs.writeFileSync('${{github.workspace}}/firebase-android.zip', Buffer.from(downloadPreview.data));
+      - run: |
+          unzip pr.zip
+          echo "pr_number=$(cat NR)" >> $GITHUB_ENV
+          mkdir firebase-android
```

Comment:
Nit: given the name of this codelab `firebase-android` seems out of place. Two options:

1. Update this directory to match the current codelab
2. Update this directory to something more generic and use it in both this and `firebase-android` codelab